src/storage/txn/process.rs

```diff
@@ -252,11 +267,24 @@ impl<E: Engine, S: MsgScheduler, L: LockManager> Executor<E, S, L> {
                         cid,
                         pr,
                         result: Ok(()),
+                        pipelined: false,
                         tag,
                     }
                 } else {
                     let sched = scheduler.clone();
                     let sched_pool = self.take_pool();
+                    let (write_finished_pr, pipelined_write_pr) = if pipelined {
```

Comment:
@sticnarf Do you have any idea to eliminate the clone here?

Reply:
Addressed

Reply:
Can we just leave an empty `write_finished_pr` if pipelined?

Reply:
No. We call `async_write` first and then notify the client because we want to make sure the write request is scheduled to the engine successfully. So it's possible the write is finished before the pipelined request.

However, I suspect the need for the order between `async_write` and `notify_scheduler`...

Reply:
It adds much complexity and the clone here.

Reply:
I don't have a good idea under the current framework...

Reply:
We can optimize it later since there are some things to do here to improve the sucess rate.